[Audio] Better error handling after multiple subsequential errors

Drapersniper opened the pull request:
### Type

- [X] Bugfix
- [X] Enhancement
- [ ] New feature

### Description of the changes

Auto disconnects the player if there are more than 5 errors in the last 10 seconds. (Likely caused by ... YouTube or playing local album with broken tracks etc)

Benefit of this is to reduce message spam caused by the bot when letting them know the playback broke

Also while at it fixed 1 bug where the DJ command was crashing

Drapersniper committed:
Merge branch 'V3/develop' of https://github.com/Cog-Creators/Red-DiscordBot into refactoring

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

# Conflicts:
#	redbot/cogs/audio/audio.py

Drapersniper committed:
:eyes: + chore

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
facepalm

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
facepalm... again y u h8 me bruh

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
fuk this fuk u tube fuck python fuck all

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
awyehfqwajefhnqeffawefqa eqewarfqaesf qwef qaf qwfr

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
fuck everything

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
oh lord saviour resus i love you just make this work

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Change logic to be no errors within last 10 seconds... this should be a valid work around discord ratelimits caused by the spam

PredaaA reviewed (dismissed):
I tested this PR and works as intended.

Drapersniper committed:
Remove auto deletion

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
black

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Migrate Playlist to DB 3 TODO
1 Migrate Config to Schema 3 without playlists
and update get_playlist methods

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Revert "Migrate Playlist to DB 3 TODO 1 Migrate Config to Schema 3 without playlists and update get_playlist methods"

This reverts commit 4af33cff

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Allow removing tracks from queue by URL

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Words matter

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
*sigh*

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
chore

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

mikeshardmind commented:
Needs conflict resolution.

Drapersniper committed:
Merge branches 'V3/develop' and 'f-yu-tube' of https://github.com/Drapersniper/Red-DiscordBot into f-yu-tube

Drapersniper committed:
Merge branch 'V3/develop' into f-yu-tube

Drapersniper committed:
arghhh CONFLICTS

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper commented:
Conflicts resolved

mikeshardmind commented on redbot/cogs/audio/audio.py:
```diff
@@ -122,7 +122,7 @@ def __init__(self, bot):
             use_external_lavalink=False,
             restrict=True,
-            current_version=redbot.core.VersionInfo.from_str("3.0.0a0").to_json(),
+            current_version=redbot.core.VersionInfo.from_str(__version__).to_json(),
```
What is this value being used for right now?

If it is or has been used for migrating data, this is not a correct change and will break anyone upgrading from the 3.0.0 -> 3.2.0 as the default here should be the lowest version, with migrations setting the version related to their migration.

Drapersniper replied:
afaik this is not currently used for anything ... but @aikaterna  should be able to confirm it already existed before I touched audio..

Drapersniper replied:
This is unused anywhere within red ... I've remove this default

Drapersniper commented:
Do not merge this one ... Merge #3205

Drapersniper committed:
Remove unused config default

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper commented:
Will setup a CPR for all of this so itsd easier to follow ... less likely to fuck up)
